python_scripts/parse_kraken_to_biom.py

```diff
@@ -0,0 +1,195 @@
+            # record abundance
+            if taxonomic_rank_level_str in taxonomy:
+                # keep taxonomy string up to specified level
+                taxonomy = taxonomy.split(split_on_level)[0]
+                sample_id = label.split('_')[0]
+                value = abundances.at[taxonomy, sample_id]
+                if np.isnan(value):
+                    abundances.set_value(taxonomy, sample_id, 1.)
+                else:
+                    abundances.set_value(taxonomy, sample_id, value+1.)
+    sample_ids = abundances.index.values.tolist()
+    for i in range(len(sample_ids)):
+        sample_ids[i] = sample_ids[i].replace("d__", "k__")
+        sample_ids[i] = sample_ids[i].replace("|", ";")
+    return Table(abundances.fillna(0.).as_matrix(),
+                 sample_ids,
+                 abundances.columns.values.tolist())
+
+
+def prepare_dataframe(kraken_translate_report_fp,
+                      taxonomic_rank,
+                      taxa_levels,
+                      taxa_levels_idx):
+    """Return sets for sample IDs and taxonomy strings.
+
+    Parameters
+    ----------
+    kraken_translate_report_fp: string
+        filepath to output of "kraken translate"
+    taxonomic_rank: string
+        taxonomy level (e.g., genus or species)
+    taxa_levels: dictionary
+        keys are full name taxonomic ranks and values are abbreviated ranks
+    taxa_levels_idx: dictionary
+        2-way dict storing integer depths for abbreviated taxonomic ranks
+
+    Returns
+    -------
+    sample_ids: list
+        all unique sample IDs in file
+    taxonomies: list
+        all unique taxonomies in file
+    """
+    total_levels = len(taxa_levels)
+    taxonomic_rank_level_str = taxa_levels[taxonomic_rank]
+    taxonomic_rank_level_int = taxa_levels_idx[taxonomic_rank_level_str]
+    if taxonomic_rank_level_int < 6:
+        split_on_level = taxa_levels_idx[str(taxonomic_rank_level_int + 1)]
+    else:
+        # keep full string (to species level)
+        split_on_level = '\t'
+    sample_ids = set()
+    taxonomies = set()
+    with open(kraken_translate_report_fp) as kraken_translate_report_f:
+        for line in kraken_translate_report_f:
+            label, taxonomy = line.strip().split('\t')
+            sample_id = label.split('_')[0]
+            sample_ids.add(sample_id)
+            # record abundance
+            if taxonomic_rank_level_str in taxonomy:
+                # keep taxonomy string up to specified level
+                taxonomy = taxonomy.split(split_on_level)[0]
+                taxonomies.add(taxonomy)
+    return list(sample_ids), list(taxonomies)
+
+
+def write_biom_table(table, biom_output_fp):
+    """Write BIOM table to file.
+
+    Parameters
+    ----------
+    table: biom.Table
+        an instance of a BIOM table
+    biom_output_fp: string
+        filepath to output BIOM table
+    """
+    with biom_open(biom_output_fp, 'w') as f:
+        if HAVE_H5PY:
```

Comment:
can this check be omitted and just require the package to depend on h5py? BIOM-format 1.0 won't die if we keep using it...